app/javascript/dashboard/components-next/Contacts/ContactsHeader/ContactHeader.vue

```diff
@@ -0,0 +1,106 @@
+<script setup>
+import Button from 'dashboard/components-next/button/Button.vue';
+import Input from 'dashboard/components-next/input/Input.vue';
+import Icon from 'dashboard/components-next/icon/Icon.vue';
+import ContactSortMenu from './components/ContactSortMenu.vue';
+import ContactMoreActions from './components/ContactMoreActions.vue';
+
+defineProps({
+  isEmptyState: {
+    type: Boolean,
+    default: false,
+  },
+  isSegmentView: {
+    type: Boolean,
+    default: true,
+  },
+  searchValue: {
+    type: String,
+    default: '',
+  },
+  headerTitle: {
+    type: String,
+    required: true,
+  },
+  buttonLabel: {
+    type: String,
+    required: true,
+  },
+  activeSort: {
+    type: String,
+    default: 'last_activity_at',
+  },
+  activeOrdering: {
+    type: String,
+    default: '',
+  },
+});
+
+const emit = defineEmits([
+  'search',
+  'filter',
+  'update:sort',
+  'message',
+  'add',
+  'import',
+  'export',
+]);
+</script>
+
+<template>
+  <header class="sticky top-0 z-10 px-6 xl:px-0">
+    <div
+      class="flex items-center justify-between w-full h-20 gap-2 mx-auto max-w-[900px]"
+    >
+      <span class="text-xl font-medium truncate text-n-slate-12">
+        {{ headerTitle }}
+      </span>
+      <div class="flex items-center flex-shrink-0 gap-4">
+        <div
+          v-if="!isEmptyState && !isSegmentView"
+          class="flex items-center gap-2"
+        >
+          <Input
+            :model-value="searchValue"
+            type="search"
+            :placeholder="$t('CONTACTS_LAYOUT.HEADER.SEARCH_PLACEHOLDER')"
+            :custom-input-class="[
+              'h-8 [&:not(.focus)]:!border-transparent bg-n-alpha-2 dark:bg-n-solid-1 ltr:!pl-8 !py-1 rtl:!pr-8',
+            ]"
+            @input="emit('search', $event.target.value)"
+          >
+            <template #prefix>
+              <Icon
+                icon="i-lucide-search"
+                class="absolute -translate-y-1/2 text-n-slate-11 size-4 top-1/2 ltr:left-2 rtl:right-2"
+              />
+            </template>
+          </Input>
+        </div>
+        <div class="flex items-center gap-2">
+          <Button
+            v-if="!isEmptyState"
+            icon="i-lucide-list-filter"
+            color="slate"
+            size="sm"
+            variant="ghost"
+            @click="emit('filter')"
+          />
+          <ContactSortMenu
+            v-if="!isEmptyState"
+            :active-sort="activeSort"
+            :active-ordering="activeOrdering"
+            @update:sort="emit('update:sort', $event)"
+          />
```

Comment:
It wouldn't hurt even if we show the sort and filter when there are no contacts. Let us skip the logic here.